salt/_modules/metalk8s_drain.py

```diff
@@ -0,0 +1,512 @@
+
+                if warning:
+                    warnings.setdefault(
+                        warning, []).append(pod.metadata.name)
+                pod_deletable = pod_deletable and filter_deletable
+            if pod_deletable:
+                pods.append(pod)
+
+        def message(errors):
+            msgs = []
+            for key, msg in errors.items():
+                msgs.append("{0}: {1}".format(key, ", ".join(msg)))
+            return "; ".join(msgs)
+
+        if failures:
+            raise DrainException(message(failures))
+        if warnings:
+            log.warning("WARNING: %s", message(warnings))
+        return pods
+
+    def run_drain(self, dry_run=False, **kwargs):
+        '''Cordon the node if necessary, then drain it.'''
+        node_unschedulable = __salt__['metalk8s_kubernetes.node_unschedulable'](
+            self.node_name, **kwargs
+        )
+        if not node_unschedulable:
+            __salt__['metalk8s_kubernetes.node_cordon'](self.node_name, **kwargs)
+        return self.evict_pods_orchestrate(dry_run=dry_run)
+
+    def evict_pods_orchestrate(self, dry_run=False):
+        '''Delete or evict pods based on eviction support.
+
+        In practice, all the versions we support have eviction, so we just
+        evict the pods.
+        '''
+        try:
+            pods = self.get_pods_for_eviction()
+        except DrainException as exc:
+            if dry_run:
+                pods = None
+            raise CommandExecutionError(
+                (
+                    "The following are not deletable: {0}. "
+                    "You can ignore DaemonSet pods with the "
+                    "ignore_daemonset flag."
+                ).format(
+                    exc.message
+                )
+            )
+
+        if dry_run:
+            return "Prepared for deletion of pods: {0}".format(
+                ", ".join(pods) if pods else "no pods to delete."
+            )
+
+        def get_pod(namespace, name):
+            '''Retrieve pod information'''
+            try:
+                api_instance = kubernetes.client.CoreV1Api()
+                api_response = api_instance.list_namespaced_pod(
+                    namespace=namespace,
+                    field_selector='metadata.name={0}'.format(name)
+                )
+                return api_response
+            except (ApiException, HTTPError) as exc:
+                if isinstance(exc, ApiException) and exc.status == 404:
+                    return None
+
+                log.exception(
+                    'Exception when calling '
+                    'CoreV1Api->list_namespaced_daemon_set')
+                raise CommandExecutionError(exc)
+
+        try:
+            self.evict_pods(pods, get_pod)
+        except DrainTimeoutException as exc:
+            raise CommandExecutionError(
+                "Drain did not complete within {0}".format(exc.message)
```

Comment:
This way of passing `self.timeout` is quite convoluted...

```suggestion
                "Drain did not complete within {} seconds".format(self.timeout)
```